blockchain/pallets/minting/src/lib.rs

```diff
@@ -182,17 +182,26 @@ pub mod pallet {
                 }
             };
 
-            if let Some(existing) = IdDatasets::<T>::get(id) {
+            let id_datasets_entry = IdDatasets::<T>::get(id);
+            if let Some(existing) = &id_datasets_entry {
                 ensure!(
-                    extension_proof.strict_extends(&existing),
+                    extension_proof.strict_extends(existing),
                     Error::<T>::ExtensionDoesNotExtendExistingDataset
                 );
             }
 
             IdDatasets::<T>::insert(id, extension_proof);
             NextMintingRewards::<T>::insert(id, who);
 
-            Ok(())
+            // Only first call is free, determined by not having an
+            // entry in .CalledRegisterForMinting. All subsequent calls
+            // will be Pays::Yes
```

Comment:
The tests explain why we have this code. I strongly think we should avoid comments in code. 

The only reason to include code comments is to explain why we're doing something in way that isn't the expected way.  I've only left code comments in two instances, one when I was calling `eval` in JavaScript, which is a potentially dangerous function, and the other when I was mutating the internals of some Java class because it gave the interface I wanted. Both of the comments were about why I was doing that instead of any alternative.

When I think that code is unclear in what it is doing or what it is for, I make a function with a name that clarifies the code. I don't think this code warrants an explanatory function, but if you do, I'm not opposed (just make sure it doesn't accidentally get exposed to users through some substrate magic).

Reply:
Yes makes sense, I will follow the coding style you prefer. Personally I comment a lot in code.